docs/ref-manual/cvl/methods.md

```diff
@@ -46,44 +55,251 @@ method_summary   ::= "ALWAYS" "(" value ")"
+declaration contains a `returns` clause, the declared return type must match
+the contract method's return type.  If the `returns` clause is omitted, the
+return type is taken from the contract method's return type.
+
+Following the `returns` clause is an optional `envfree` tag.  Marking a method
+with `envfree` has two effects.  First, {ref}`calls <call-expr>` to the method
+from CVL do not need to explicitly pass an {term}`environment` value as the
+first argument.  Second, the prover will verify that the method implementation
+in the contract being verified does not depend on any of the environment
+variables.  The results of this check are displayed on the verification report
+as separate rules called `envfreeFuncsStaticCheck` and
+`envfreeFuncsAreNonpayable`[^envfree_nonpayable].
+
+[^envfree_nonpayable]: The effect of payable functions on the contract's
+  balance depends on the message value, so payable functions also require an
+  `env`.
+
+Finally, the method entry may contain an optional summarization (indicated by
+`=>` followed by the summary type and an optional application policy).  A
+summarized declaration indicates that the prover should replace some calls to
+the summarized function by an approximation.  This is an important technique
+for working around prover timeouts and also for working with external contracts
+whose implementation is not fixed at verification time.
+
+The summary type determines what type of approximation is used to replace the
+function calls.  The available types are described in the following sections:
+
+ * {ref}`const-summary`
+ * {ref}`havoc-summary`
+ * {ref}`dispatcher`
+ * {ref}`auto-summary`
+ * {ref}`function-summary`
+
+The application policy determines which function calls are replaced by
+approximations.  See {ref}`summaries` for details.
+
+(summaries)=
+Which function calls are summarized
+-----------------------------------
+
+Method summaries apply to all calls with the matching ABI signature, including
+internal methods and external methods on all contracts.  There is currently no
+way to apply different summaries to different contracts or to summarize some
+calls and not others to methods with the same ABI signature.  For this reason,
+it is not possible to specify a summary for a method that is qualified by a
+contract name.
+
+The Solidity compiler allows internal functions to have richer type
+specifications than just ABI specifications, but the prover only considers the
+ABI encoding to determine whether a method call should be summarized.  For
+example, internal functions can accept struct arguments, but the structs are
+replaced by tuples in the ABI signature.  If the generated ABI signature of an
+internal method call matches the ABI signature specified in the methods block,
+it will be a candidate for summarization even if the declared argument types in
+the Solidity function don't exactly match the declared argument types in the
+`methods` block.
```

Comment:
@thomasbernardi I thought this was no longer the case

Reply:
We should discuss and maybe look at tests, because this was my understanding of what Thomas explained to me

Reply:
resolved in latest draft